#! /bin/sh -e
## 01_upsstatsCgiNoHostCrash.dpatch by <aquette@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: NUT clean rules misses the removal of clients/libupsclient.a (#283539).

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0
            ;;
    -unpatch) patch -R -p1 ${patch_opts} < $0 
              ;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

--- nut-2.0.1.orig/clients/upsstats.c	2003-09-30 00:50:10.000000000 -0400
+++ nut-2.0.1/clients/upsstats.c	2005-04-03 12:49:39.583284972 -0400
@@ -752,6 +752,20 @@ static void load_hosts_conf(void)
 	}
 
 	pconf_finish(&ctx);
+
+	if (!ulhead) {
+		printf("<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0 Transitional//EN\"\n");
+		printf("	\"http://www.w3.org/TR/REC-html40/loose.dtd\">\n");
+		printf("<HTML><HEAD>\n");
+		printf("<TITLE>Error: no hosts to monitor</TITLE>\n");
+		printf("</HEAD><BODY>\n");
+		printf("Error: no hosts to monitor (check <CODE>hosts.conf</CODE>)\n");
+		printf("</BODY></HTML>\n");
+
+		/* leave something for the admin */
+		fprintf(stderr, "upsstats: no hosts to monitor\n");
+		exit(EXIT_FAILURE);
+	}
 }
 
 static void display_single(void)

